### PR TITLE
Check for required python libraries for creating registry

### LINKED ISSUE #358: ansible playbook missing python libraries
```
TASK [installer : Generate an OpenSSL private key] *****************************
Tuesday 05 May 2020  06:57:11 +0000 (0:00:00.375)       0:01:52.894 *********** 
fatal: [openshift-worker-0.fci1.kni.lab.eng.bos.redhat.com -> openshift-worker-0.fci1.kni.lab.eng.bos.redhat.com]: FAILED! => {"changed": false, "msg": "Can't detect any of the required Python libraries cryptography (>= 1.2.3) or PyOpenSSL (>= 0.6)"}
```

this is not listed in prerequisits / installed by playbook

### PR DESCRIPTION
**Pull request by iranzo**
# Description

During validations if registry creation is needed, fail on missing dependency alternatives

Closes #358 

## Type of change

Please select the appropiate options:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Testing

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:

- Versions:
- Hardware:

## Checklist

- [x] I have performed a self-review of my own code
- [ ] If a change is adding a feature, it should require a change to the README.md and the review should catch this.
- [ ] If the change is a fix, it should have an issue. The review should make sure the comments state the issue (not just the number) and it should use the keywords that will close the issue on merge.
- [x] A change should not being merged unless it passes CI or there is a comment/update saying what testing was passed.
- [x] PRs should not be merged unless positively reviewed.

### CODE REVIEW

File: ansible-ipi-install/roles/node-prep/tasks/10_validation.yml [hunk 1]
```diff
@@ -22,6 +22,30 @@
   - always
   - validation
 
+- name: Ensure Python cryptography libraries are installed
+  python_requirements_info:
+    dependencies:
+    - cryptography>=1.2.3
+  delegate_to: localhost
+  register: _py_crypto
+  ignore_errors: yes
```
Comment (rlopez133) [inline ansible-ipi-install/roles/node-prep/tasks/10_validation.yml:31]: you will want the tags: validation in here as well

File: ansible-ipi-install/roles/node-prep/tasks/10_validation.yml [hunk 2]
```diff
@@ -22,6 +22,30 @@
...
+  register: _py_pyopenssl
+  ignore_errors: yes
+
+- name: "Fail on missing required cryptography libraries cryptography OR PyOpenSSL"
+  fail:
+    msg: "Required cryptography libraries are missing cryptography OR PyOpenSSL"
+  when: (_py_crypto.not_found != []) and
+        (_py_pyopenssl.not_found != [] )
+  tags:
+    - create_registry
```
Comment (rlopez133) [inline ansible-ipi-install/roles/node-prep/tasks/10_validation.yml:47]: you will want the tags: validation in here as well

File: ansible-ipi-install/roles/node-prep/tasks/10_validation.yml [hunk 3]
```diff
@@ -22,6 +22,30 @@
...
+  register: _py_crypto
+  ignore_errors: yes
+
+- name: Ensure Python PyOpenSSL libraries are installed
+  python_requirements_info:
+    dependencies:
+    - PyOpenSSL>=0.6
+  delegate_to: localhost
+  register: _py_pyopenssl
+  ignore_errors: yes
```
Comment (rlopez133) [inline ansible-ipi-install/roles/node-prep/tasks/10_validation.yml:39]: you will want the tags: validation in here as well

File: ansible-ipi-install/roles/node-prep/tasks/10_validation.yml (hunk 2 above)
Comment (iranzo) [reply ansible-ipi-install/roles/node-prep/tasks/10_validation.yml:47]: But we only want to check if libraries are installed in the case we want to create a registry, and ignore it for any other case, isn't it?

**Comment by iranzo** [comment]
[test-ipi]

**Pull request merged**